### PR TITLE
Add /vanish and /invsee

### PR DESCRIPTION
**Pull request by ghost**
PR for issues #2022 and #2018
Enjoy!

### CODE REVIEW

**Comment by mayokek** [comment]
As per the contributing guidelines, you may make a PR that solves an issue, not multiple.
```
Due to a lot of pull requests, and the effort it takes to review each PR, some requirements for pull requests have been imposed. PRs opened that do not meet the pull request requirements may be closed.

PRs must only address issues which are marked as "accepted".
PRs must only address one issue per PR, not several.
PRs must not solely be grammar fixes, config-edits, small changes (less than 10 lines of code) or message rephrasing edits. Small edits such as these take very little effort to make and a lot more effort to review and integrate into the codebase.
PRs must be compiled and tested before they are submitted. If your PR does not function or has clearly not been tested, it may be closed without notice.
```

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 1]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
         }
         return false;
     }
+
+    public Boolean isVanished(String username)
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:141]: Use lowercase b

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 2]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+        {
+            return user.isVanished();
+        }
+
+        return null;
+    }
+
+    public void setVanished(String username, boolean vanished)
+    {
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 3]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+        final IUser user = getEssentialsUser(username);
+        if (user != null)
+        {
+            user.setVanished(vanished);
+        }
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            }
+            plugin.esb.setVanished(playerSender.getName(), true);
+            playerSender.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, 1000000, 1000000, true, false));
+            fPlayer.setVanish(true);
+            return true;
+        }
+
+        return true;
+    }
+}
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java]: Missing an extra line

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 5]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+                }
+            }
+        }
+
+        if (refreshPlayer != null)
+        {
+            final Player player = refreshPlayer;
+            new BukkitRunnable()
+            {
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 6]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+            fPlayer.setInvsee(false);
+            refreshPlayer = playerdata;
+        }
+
+        if (refreshPlayer != null)
+        {
+            final Player player = refreshPlayer;
+            new BukkitRunnable()
+            {
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 7]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+                @Override
+                public void run()
+                {
+                    player.updateInventory();
+                }
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 8]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onPlayerQuit(PlayerQuitEvent event)
+    {
+        final Player player = event.getPlayer();
+        final FPlayer fPlayer = plugin.pl.getPlayer(player);
+        if (fPlayer.isVanish())
+        {
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+@CommandPermissions(level = Rank.OP, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Look into another player's inventory, optionally take items out.", usage = "/<command> <player>", aliases = "inv,insee")
+public class Command_invsee extends FreedomCommand
+{
+
+    @Override
+    public boolean run(CommandSender sender, Player playerSender, Command cmd, String commandLabel, String[] args, boolean senderIsConsole)
+    {
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java:18]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        {
+            msg("You cannot invsee yourself.");
+            return true;
+        }
+
+        if (isAdmin(player) && !isAdmin(playerSender))
+        {
+            msg("You can't spy on admins!");
+            return true;
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        playerSender.closeInventory();
+        FPlayer fPlayer = plugin.pl.getPlayer(playerSender);
+        fPlayer.setInvsee(true);
+        Inventory playerInv = player.getInventory();
+        playerSender.openInventory(playerInv);
+        return true;
+    }
+
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: White line should be removed from here.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+}
```
Comment (mayokek) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: Missing an extra line after this

**Comment by mayokek** [comment]
I should've use review changes instead of leaving single-comment

**Comment by ghost** [comment]
mayokek, no offence but that review is completely useless. We are more relaxed about this stuff now and an extra line of whitespace does not hurt, neither does putting two very similar issues into one PR.

**Comment by mayokek** [comment]
Formatting matters when you are coding. And also, how does those 2 issues similar?

**Comment by ghost** [comment]
There is nothing wrong with the formatting.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java (hunk 1 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:141]: No, this is correct.

Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java:141]: Quoting stackoverflow: "boolean is a java primitive type whereas Boolean is an object/reference type that wraps a boolean" To return null it has to be an Object, because boolean can only return true or false.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 13]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+        final IUser user = getEssentialsUser(username);
+        if (user != null)
+        {
+            user.setVanished(vanished);
+        }
+
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: This should probably be moved into a listener class and not in the bridge its self.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 14]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+                public void run()
+                {
+                    player.updateInventory();
+                }
+
+            }.runTaskLater(this.plugin, 20);
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Again, this should be in a listener file really.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 15]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Again, should be in a listener file.

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 16]
```diff
@@ -136,4 +152,131 @@ public boolean isEssentialsEnabled()
...
+        {
+            final FPlayer fPlayer = plugin.pl.getPlayer(vanishedPlayers);
+            if (fPlayer.isVanish())
+            {
+                player.hidePlayer(vanishedPlayers);
+            }
+        }
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Also in a listener file

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java (hunk 12 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: What do you mean "Missing an extra line"?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+import org.apache.commons.lang.StringUtils;
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+@CommandPermissions(level = Rank.SUPER_ADMIN, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Vanish yourself.", usage = "/<command>", aliases = "v")
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java]: Not sure we want aliases... Im sure we had the discussion before and wanted to avoid them in order to keep things simple.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+
+import me.totalfreedom.totalfreedommod.player.FPlayer;
+import me.totalfreedom.totalfreedommod.rank.Rank;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+
+@CommandPermissions(level = Rank.OP, source = SourceType.ONLY_IN_GAME)
+@CommandParameters(description = "Look into another player's inventory, optionally take items out.", usage = "/<command> <player>", aliases = "inv,insee")
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: Also not sure we wanted aliases. Especially not multiple ones.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            msg(ChatColor.GOLD + "You have been vanished.");
+            FUtil.bcastMsg(playerSender.getName() + " left the game", ChatColor.YELLOW);
+            FLog.info(playerSender.getName() + " is now vanished.");
+            for (Player player : server.getOnlinePlayers())
+            {
+                player.hidePlayer(playerSender);
+            }
+            plugin.esb.setVanished(playerSender.getName(), true);
+            playerSender.addPotionEffect(new PotionEffect(PotionEffectType.INVISIBILITY, 1000000, 1000000, true, false));
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java]: Is there not a better way of doing this? Like GameMode 3?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java [hunk 20]
```diff
@@ -0,0 +1,70 @@
+package me.totalfreedom.totalfreedommod.command;
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:1]: You should probably also update the /list command, as invis players should still show in that because TFM will not check if they are invis currently.

**Review by Wild1145 — CHANGES_REQUESTED** [review]

**Comment by Wild1145** [comment]
More generally in regards to formatting, while its not the end of the world, if you configure your IDE once properly, you can just format the files as you're working on them, and before you commit them. Formatting will need to be corrected, but I think given these are smaller changes we can be flexible in not needing two different PR's on this occasion.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java (hunk 12 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: Unless I am mistaken, At the end of the code, there always a new line.

Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: There shouldnt be generally.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 21]
```diff
@@ -33,18 +31,6 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
         {
             return false;
         }
-
-        if (FUtil.isFromHostConsole(sender.getName()))
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: Why did you remove this?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java (hunk 12 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: You're correct.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 22]
```diff
@@ -68,31 +54,54 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
             listFilter = ListFilter.PLAYERS;
         }
 
+        if (listFilter == ListFilter.VANISHED_ADMINS && !isAdmin(playerSender))
+        {
+            msg("You can't view vanished admins.");
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: I probably wouldnt even list this, I'd just make it return an unkown command with the syntax.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java (hunk 19 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java]: The whole idea of this command is to completely vanish but retain their current gamemode, if users wish to do this then they can go into that gamemode themselves

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 23]
```diff
@@ -2,27 +2,25 @@
...
-import me.totalfreedom.totalfreedommod.util.FUtil;
 import org.apache.commons.lang3.StringUtils;
 import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandSender;
 import org.bukkit.entity.Player;
 
 @CommandPermissions(level = Rank.IMPOSTOR, source = SourceType.BOTH)
-@CommandParameters(description = "Lists the real names of all online players.", usage = "/<command> [-a | -i | -f]", aliases = "who")
+@CommandParameters(description = "Lists the real names of all online players.", usage = "/<command> [-a | -i | -v]", aliases = "who")
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: Not sure I'd add -v to the syntax, if only admins know invis admins are a thing its no bad thing.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java (hunk 20 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:1]: Done

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 24]
```diff
@@ -2,27 +2,25 @@
...
 public class Command_list extends FreedomCommand
 {
 
     private static enum ListFilter
     {
-
         PLAYERS,
         ADMINS,
-        FAMOUS_PLAYERS,
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: Why remove famous players? While I didnt know it was a thing, im not sure why we're removing the functionality?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java (hunk 21 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: oh rip, wrong code

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 25]
```diff
@@ -68,31 +54,54 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
...
         for (Player player : server.getOnlinePlayers())
         {
-            if (listFilter == ListFilter.ADMINS && !plugin.al.isAdmin(player))
+            FPlayer fPlayer = plugin.pl.getPlayer(player);
+            if (listFilter == ListFilter.ADMINS && !isAdmin(player))
             {
                 continue;
             }
 
-            if (listFilter == ListFilter.IMPOSTORS && !plugin.al.isAdminImpostor(player))
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java]: Why remove imposters from the list?

**Review by Wild1145 — CHANGES_REQUESTED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+public class Invsee extends FreedomService
+{
+
+    public Invsee(TotalFreedomMod plugin)
+    {
+        super(plugin);
+    }
+
+    @Override
+    protected void onStart()
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:25]: Im not sure why you need this, it does nothing?

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+    {
+
+    }
+
+    @Override
+    protected void onStop()
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:31]: Im not sure why you need this, it does nothing?

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java (hunk 2 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Agreed

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java (hunk 3 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Agreed

File: src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java [hunk 28]
```diff
@@ -136,4 +137,28 @@ public boolean isEssentialsEnabled()
...
+    {
+
+        final IUser user = getEssentialsUser(username);
+        if (user != null)
+        {
+            user.setVanished(vanished);
+        }
+
+    }
+
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/bridge/EssentialsBridge.java]: Also if you could remove this line as well

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java (hunk 10 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: Agreed

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java [hunk 29]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        if (isAdmin(player) && !isAdmin(playerSender))
+        {
+            msg("You can't spy on admins!");
+            return true;
+
+        }
+
+        playerSender.closeInventory();
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: Why are you closing the original players inventory? Does it then not give it away that you're looking in it?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java [hunk 30]
```diff
@@ -101,7 +134,8 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
         String playerType = listFilter == null ? "players" : listFilter.toString().toLowerCase().replace('_', ' ');
 
         onlineUsers.append("Connected ");
-        onlineUsers.append(playerType + ": ");
+        onlineUsers.append(playerType);
+        onlineUsers.append(": ");
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:138]: Why have you split this into two lines?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java (hunk 19 above)
Comment (Wild1145) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java]: I guess my concern is this is not completely vanished though. Players can still be seen with this both through hacked clients and having particle effects enabled?

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java [hunk 31]
```diff
@@ -0,0 +1,70 @@
...
+        Admin admin = getAdmin(playerSender);
+        final FPlayer fPlayer = plugin.pl.getPlayer(playerSender);
+        if (fPlayer.isVanish())
+        {
+            msg(ChatColor.GOLD + "You have been unvanished.");
+            if (admin.hasLoginMessage())
+            {
+                loginMsg = FUtil.colorize(admin.getLoginMessage());
+            }
+            FUtil.bcastMsg(ChatColor.AQUA + playerSender.getName() + " is " + loginMsg);
```
Comment (Wild1145) [inline src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:39]: Im not sure on this. Is this not different depending on the rank / title the admin has?

**Review by Wild1145 — DISMISSED** [review]

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java (hunk 26 above)
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:25]: Aero requires it, I think, for services.

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java (hunk 27 above)
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:31]: Aero requires it, I think, for services.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java (hunk 29 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_invsee.java]: @Wild1145, the `playerSender` method is still sender that have been converted to player.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java (hunk 31 above)
Comment (lemonsked) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_vanish.java:39]: Nope, it isn't.

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java (hunk 26 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:25]: Lemon is correct. `FreedomService` requires the method to be added

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java (hunk 27 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:31]: Lemon is correct. `FreedomService` requires the method to be added

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java (hunk 30 above)
Comment (ghost) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:138]: Because you don't concatenate in .append, you append it.

**Comment by ghost** [comment]
All requested changes have been implemented

**Comment by mayokek** [comment]
@Commodore64x, I don't believe spectator is a good idea because if we need to interact with items, we must go out of vanish.

**Comment by ghost** [comment]
Wild asked for it to be changed, regardless you can go into creative anyway

**Comment by Wild1145** [comment]
@Commodore64x @mayokek Spectator may not be the best way to implement it, the honest answer is I dont know what the absolute best way is. Spectator sounds like it'll be the best because its totally invisible. You should not need to interact with blocks and similar when in /invis anyway??

**Comment by Wild1145** [comment]
@Commodore64x Have I misunderstood the purpose of /invis?

**Comment by Wild1145** [comment]
@Commodore64x Sorry, My mistake, I did not mean /invis but /vanish - My mistake, its been a long day. I was in fact referring to /vanish when I was asking about people being made invisible.

**Comment by ghost** [comment]
@Wild1145 Can you remove the Needs Changes tag? There are no outstanding changes that haven't been fixed.

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java [hunk 32]
```diff
@@ -0,0 +1,106 @@
...
+
+    public Invsee(TotalFreedomMod plugin)
+    {
+        super(plugin);
+    }
+
+    @Override
+    protected void onStart()
+    {
+
```
Comment (OxLemonxO) [inline src/main/java/me/totalfreedom/totalfreedommod/Invsee.java]: Remove whitespace.

Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:27]: Remove whitespace.

File: src/main/java/me/totalfreedom/totalfreedommod/Invsee.java [hunk 33]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    @Override
+    protected void onStop()
+    {
+
```
Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/Invsee.java:33]: Remvoe whitespace.

File: src/main/java/me/totalfreedom/totalfreedommod/Vanish.java [hunk 34]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @EventHandler(priority = EventPriority.MONITOR)
+    public void onPlayerQuit(PlayerQuitEvent event)
+    {
+        final Player player = event.getPlayer();
+        final FPlayer fPlayer = plugin.pl.getPlayer(player);
+        if (fPlayer.isVanish())
+        {
+
```
Comment (lemonsked) [inline src/main/java/me/totalfreedom/totalfreedommod/Vanish.java:54]: Remove extra whitespace.

File: src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java (hunk 30 above)
Comment (mayokek) [reply src/main/java/me/totalfreedom/totalfreedommod/command/Command_list.java:138]: This can be merge into one line.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
This is currently not building for me. Can you take a look at this for me please?

**Comment by Wild1145** [comment]
@Commodore64x This is still failing the TravisCI Integration tests.

**Comment by lemonsked** [comment]
@Wild1145  For some reason, the TravisCI integration tests think that the language is ruby, so it fails.
https://travis-ci.org/TotalFreedom/TotalFreedomMod/jobs/257602861/config

**Comment by Wild1145** [comment]
@OxLemonxO That'd do it.

**Comment by JeromSar** [comment]
@Wild1145 @OxLemonxO The TotalFreedomMod `devel` branch now contains the required _.travis.yml_ file for correct Travis CI integration. @Commodore64x Merge this branch with the upstream `devel` branch to fix the issue.

**Comment by ghost** [comment]
@JeromSar @Wild1145 Should work with travis now.

**Comment by Wild1145** [comment]
@Commodore64x Im not sure what you've done, but you've done something really weird with the Git History there...
